ytsync/ytsync.go

```diff
@@ -65,12 +65,24 @@ type Sync struct {
 	claimAddress   string
 	videoDirectory string
 	db             *redisdb.DB
+	syncedVideos   map[string]syncedVideo
+	grp            *stop.Group
+	lbryChannelID  string
+
+	videosMapMux sync.Mutex
+	mux          sync.Mutex
+	wg           sync.WaitGroup
```

Comment:
this should also be a pointer. but more importantly, you don't need this if you already have a stop.Group. stop.Group works as a combo WaitGroup + channel that will be closed to indicate stopping (and can be closed safely multiple times). so use `grp.Add()` and `grp.Wait()` instead

Reply:
it won't let me comment below, so I'm commenting here:

you call `Add()` and `Done()` inside `startWorker(workerNum int)`, but the correct pattern is to make those calls outside the function. startWorker() doesn't know if its being run asynchronously or not. you only need a waitgroup if it is. there are also subtle concurrency issues with calling it inside the function. so the right way to go is to remove Add and Done from inside startWorker, and do this:

```
	for i := 0; i < s.ConcurrentVideos; i++ {
		s.wg.Add(1)
		go func() {
			defer s.wg.Done()
			s.startWorker(i)
		}()
	}
```

Reply:
I need to look into that, this is from your original code and I don't remember ever changing it. Thanks for the pointers there.

Reply:
I looked more into why we need to lock the whole walletSetup function. The reason I'm doing that is to avoid multiple threads from refilling the wallet concurrently. I don't think I can easily break up the the function to lock fewer lines of code. I think it's fair to leave it like that.

I removed the wait group in favor of the stop group